File: AtomykPlay/app/src/main/java/com/atomykcoder/atomykplay/adapters/Generics/GenericRecyclerAdapter.java
package com.atomykcoder.atomykplay.adapters.Generics;

import android.annotation.SuppressLint;
import android.os.SystemClock;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class GenericRecyclerAdapter<T> extends RecyclerView.Adapter<GenericViewHolder<T>> {

    protected ArrayList<T> items;
    protected int delay = 600;
    long lastClickTime;

    @SuppressLint("NotifyDataSetChanged")
    public void addItems(ArrayList<T> _items) {
        items.addAll(_items);
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public GenericViewHolder<T> onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View view = inflater.inflate(viewType, parent, false);
        return new GenericViewHolder<T>(view);
    }

    @Override
    public void onBindViewHolder(@NonNull GenericViewHolder<T> holder, int position) {
        holder.onBind(items.get(position));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    protected boolean shouldIgnoreClick() {
        if (SystemClock.elapsedRealtime() < (lastClickTime + delay)) {
            return true;
        }
        lastClickTime = SystemClock.elapsedRealtime();
        return false;
    }
}

File: AtomykPlay/app/src/main/java/com/atomykcoder/atomykplay/adapters/generics/GenericRecyclerAdapter.kt
package com.atomykcoder.atomykplay.adapters.generics

import android.annotation.SuppressLint
import android.os.SystemClock
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import java.util.ArrayList

open class GenericRecyclerAdapter<T> : RecyclerView.Adapter<GenericViewHolder<T>>() {
    @JvmField
    protected var items: ArrayList<T>? = null
    private var delay = 600
    private var lastClickTime: Long = 0

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): GenericViewHolder<T> {
        val inflater = LayoutInflater.from(parent.context)
        val view = inflater.inflate(viewType, parent, false)
        return GenericViewHolder(view)
    }

    override fun onBindViewHolder(holder: GenericViewHolder<T>, position: Int) {
        items?.get(position)?.let { holder.onBind(it) }
    }

    override fun getItemCount(): Int {
        return items?.size ?: 0
    }

    protected fun shouldIgnoreClick(): Boolean {
        if (SystemClock.elapsedRealtime() < lastClickTime + delay) {
            return true
        }
        lastClickTime = SystemClock.elapsedRealtime()
        return false
    }
}